plugin.video.alfa/channels/cuevana2.py

```diff
@@ -0,0 +1,290 @@
+
+def by(item, pattern):
+    logger.info()
+    itemlist = []
+
+    #descarga la pagina html
+    data = load_data(item.url)
+
+    #patron para buscar en la pagina
+    pattern = '<li class="cat-item cat-item-\d+"><a href="([^"]+)" >&&</a>'.replace('&&', pattern)
+
+    matches = scrapertools.find_multiple_matches(data, pattern)
+    for link, genre in matches:
+        itemlist.append(Item(channel = __channel__, title=genre, url=link, action="movies"))
+
+    return itemlist
+
+def genre(item):
+    return by(item, '(\D+)')
+
+def age(item):
+    return by(item, '(\d+)')
+
+def GKPluginLink(hash):
+    re = requests.post('https://player4.cuevana2.com/plugins/gkpluginsphp.php', dict(link=hash))
+
+    return re.json()['link'] if re.content else ''
+
+#el pattern esta raro para eliminar los duplicados, de todas formas asi es un lenguaje de programacion verificando su sintaxis
+def getContentMovie(data, item):
+    item.infoLabels["year"] = scrapertools.find_single_match(data, 'rel="tag">(\d+)</a>')
+    genre = ''
+    for found_genre in scrapertools.find_multiple_matches(data, 'genero/.*?">(.*?)</a>(?=.*?</p>)'):
+        genre += found_genre + ', '
+    item.infoLabels["genre"] = genre.strip(', ')
+
+    director = ''
+    for found_director in scrapertools.find_multiple_matches(data, 'director/.*?">(.*?)</a>(?=.*?</p>)'):
+        director += found_director + ', '
+    item.infoLabels["director"] = director.strip(', ')
+
+    item.infoLabels["cast"] = tuple(found_cast for found_cast in scrapertools.find_multiple_matches(
+        data, 'reparto/.*?">(.*?)</a>(?=.*?</p>)'))
+
+def getContentShow(data, item):
+    item.thumbnail = scrapertools.find_single_match(data, 'width="120" height="160" src="([^"]+)"')
+    item.infoLabels['genre'] = scrapertools.find_single_match(data, '-4px;">(.*?)</div>')
+
+def findvideos(item):
+    logger.info()
+    itemlist = []
+
+    data = load_data(item.url)
+    if item.extra:
+        getContentShow(data, item)
+    else:
+        getContentMovie(data, item)
+    pattern = '<iframe width="650" height="450" scrolling="no" src="([^"]+)'
+    subtitles = scrapertools.find_single_match(data, '<iframe width="650" height="450" scrolling="no" src=".*?sub=([^"]+)"')
+
+    #itemlist.append(Item(channel = __channel__, title=item.url))
+    for link in scrapertools.find_multiple_matches(data, pattern):
+        #php.*?=(\w+)&
+        #url=(.*?)&
+        if 'player4' in link:
+            if r'ir.php' in link:
+                link = scrapertools.find_single_match(link, 'php\?url=(.*?)&').replace('%3A', ':').replace('%2F', '/')
+                logger.info("CUEVANA IR %s" % link)
+            elif r'gdv.php' in link:
+                # google drive hace lento la busqueda de links, ademas no es tan buena opcion y es el primero que eliminan
+                continue
+            else:
+                link = scrapertools.find_single_match(link, 'php.*?=(\w+)&')
+                link = GKPluginLink(link)
```

Comment:
Hola @mrgaturus , no deberías usar la librería request por que puede llegar a ser incompatible con otras plataformas.
En su lugar deberías usar el parámetro post de downloadpage de la siguiente manera:

post = urllib.urlencode({"link":link}) 
j_data = httptools.downloadpage("https://player4.cuevana2.com/plugins/gkpluginsphp.php", post = post).data
link = jsontools.load(j_data)["link"]


Y agregar estos import:
import urllib
from core import jsontools

Esos códigos son mas compatibles en diversas plataformas.

En la película: Ibiza hay un link directo que usa el servidor: chomikuj.pl

Este link necesita agregar un referer al enlace, porque de lo contrario da error: 401

ERROR: CCurlFile::FillBuffer - Failed: HTTP returned error 401

`                if "chomikuj.pl" in link:
                    link += "|Referer=https://player4.cuevana2.com/plugins/gkpluginsphp.php"
`
Con éste referer ya funciona el link de chomikuj.pl.

Saludos.

Reply:
Hola @Intel11, Hasta ahora usando requests me ha funcionado excelente en Windows, Linux y Android. pero se agradece que httptools tiene algo para manejar el Post y sobre chomikuj.pl, el que sale como formato aspx, que raro, ese funciona muy bien en todos los sitios que lo he probado (2 TV Box y mi PC), ya lo adapto a esa manera entonces